First commit

constantineas opened the pull request:
Первая попытка сдачи проекта 4го Спринта

constantineas committed:
First commit

mkouznetsov-git commented on src/test/java/com/example/spintsixproject/LandingPageTest.java:
```diff
@@ -0,0 +1,77 @@
+    // Параметризированный тест проверки списка "Вопросы о важном"
+    @Test
+    public void questionsMenu() {
+        String expectedText = qaText;
```
Лишняя переменная `expectedText`, можно ведь сразу подставлять `qaText` в assertEquals.

constantineas replied:
Удалил лишнюю переменную expectedText

mkouznetsov-git commented on src/test/java/com/example/spintsixproject/ScooterOrderTest.java:
```diff
@@ -0,0 +1,136 @@
+    {
+        return new Object[][]
+                { // передали тестовые данные
+                        {"Костя", "Асалин", "Москва", "сокольники", "+79636563763", "30.10.2022", "Позвонить за час", "Заказ оформлен"},
```
Чем принципиально отличаются эти два набора тестовых данных? Для чего их два?

constantineas replied:
Изменил тестовые наборы, добавил с вариантом пустой комментарий и другоим форматом телефона

mkouznetsov-git commented on src/test/java/com/example/spintsixproject/ScooterOrderTest.java:
```diff
@@ -0,0 +1,136 @@
+
+    // Проверка заказа самоката по кнопке внизу
+    @Test
+    public void testClickDownButton() throws InterruptedException {
```
Это называется дублирование своего кода, и лучше этого не допускать -- у тебя уже есть точно такой же тест, который запускается по другой кнопке. Параметризуй кнопку и используй один тестовый метод. )

constantineas replied:
Параметризировал тесты заказов. Изменил метод в классе LandingPage для заказов по верхней и нижней кнопке, теперь это один метод с именем clickOrderButton(int).

constantineas committed:
1. Удалил лишнюю переменную expectedText
2. Изменил тестовые наборы, добавил с вариантом пустой комментарий и другоим форматом телефона
3. Параметризировал тесты заказов. Изменил метод в классе LandingPage для заказов по верхней и нижней кнопке, теперь это один метод с именем clickOrderButton(int).

mkouznetsov-git commented on src/test/java/Constants/Const.java:
```diff
@@ -0,0 +1,14 @@
+    // Константа для времени таймаута
+    public static final int TIME_OUT_IN_SECONDS = 10;
+
+    public static final int BUTTON_UP = 0;
```
Хорошо, что создал константы для кнопок, это правильно.

mkouznetsov-git commented on src/test/java/com/example/spintsixproject/ScooterOrderTest.java:
```diff
@@ -0,0 +1,110 @@
+                        {"Костя", "Асалин", "Москва", "сокольники", "+79636563763", "30.10.2022", "Позвонить за час", "Заказ оформлен", Const.BUTTON_UP},
+                        {"Артем", "Тарасов", "Москва", "черкизовская", "89853654102", "29.10.2022", "", "Заказ оформлен", Const.BUTTON_UP},
+                        {"Костя", "Асалин", "Москва", "сокольники", "+79636563763", "30.10.2022", "Позвонить за час", "Заказ оформлен", Const.BUTTON_DOWN},
+                        {"Артем", "Тарасов", "Москва", "черкизовская", "89853654102", "29.10.2022", "", "Заказ оформлен", Const.BUTTON_DOWN},
```
Правильно, что ты здесь передаёшь константы кнопок. Но вот в функции ты сравниваешь с 0 и 1, а надо бы -- с константами.

mkouznetsov-git commented on src/test/java/pageObject/LandingPage.java:
```diff
@@ -0,0 +1,83 @@
+
+    // Метод для нажатия по кнопке Заказать
+    public void clickOrderButton(int x){
+        if (x == 0) {
```
Вот это надо сравнивать с константами, а не с 0 и 1. Это и безопаснее, если тебе вдруг понадобится поменять значения констант, и более понятно для стороннего читателя твоего кода.

constantineas replied:
Поправил сравнение с цифрами, поменял на сравнение с константами.

mkouznetsov-git commented on src/test/java/pageObject/LandingPage.java:
```diff
@@ -0,0 +1,83 @@
+    }
+
+    // Метод для нажатия по кнопке Заказать
+    public void clickOrderButton(int x){
```
Переменные следует называть понятно, иначе не ясно, о чём идёт речь -- какой-то магический икс...

constantineas replied:
Поправил имя параметра в методе clickOrderButton() на понятный

constantineas committed:
Поправил сравнение с цифрами, поменял на сравнение с константами.

Поправил имя параметра в методе clickOrderButton() на понятный
